app/src/main/java/com/swent/mapin/ui/settings/SettingsViewModel.kt

```
@@ -0,0 +1,203 @@
+package com.swent.mapin.ui.settings
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.firestore.FirebaseFirestore
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+/** Data class representing map preferences */
+data class MapPreferences(
+    val showPOIs: Boolean = true,
+    val showRoadNumbers: Boolean = true,
+    val showStreetNames: Boolean = true,
+    val enable3DView: Boolean = false
+)
+
+/**
+ * ViewModel for managing settings state and operations.
+ *
+ * Responsibilities:
+ * - Manage map preferences/visibility toggles
+ * - Handle logout
+ * - Handle account deletion
+ * - Persist settings to Firestore
+ */
+class SettingsViewModel(
+    private val auth: FirebaseAuth = FirebaseAuth.getInstance(),
+    private val firestore: FirebaseFirestore = FirebaseFirestore.getInstance()
+) : ViewModel() {
+
+  // Map preferences state
+  private val _mapPreferences = MutableStateFlow(MapPreferences())
+  val mapPreferences: StateFlow<MapPreferences> = _mapPreferences.asStateFlow()
+
+  // Loading state for async operations
+  private val _isLoading = MutableStateFlow(false)
+  val isLoading: StateFlow<Boolean> = _isLoading.asStateFlow()
+
+  // Error state for displaying error messages
+  private val _errorMessage = MutableStateFlow<String?>(null)
+  val errorMessage: StateFlow<String?> = _errorMessage.asStateFlow()
+
+  init {
+    // Load saved preferences from Firestore
+    loadMapPreferences()
+  }
+
+  /** Load map preferences from Firestore */
+  private fun loadMapPreferences() {
+    viewModelScope.launch {
+      try {
+        val userId = auth.currentUser?.uid
+        if (userId != null) {
```

Question: No handling for the case when the user is not authenticated (userId is null). Is this scenario covered by tests? Should the ViewModel emit an explicit error message or state for this case, or is silent fallback to default preferences acceptable?